src/queryResult/sqlCodeLensProvider.ts

```diff
@@ -32,25 +33,42 @@ export class SqlCodeLensProvider implements vscode.CodeLensProvider, vscode.Disp
         if (!shouldShowActiveConnection) {
             return [];
         }
-        const connection = this._connectionManager.getConnectionInfo(document.uri.toString(true));
 
-        const items: vscode.CodeLens[] = [
-            new vscode.CodeLens(new vscode.Range(0, 0, 0, 0), {
-                title: connection
-                    ? generateServerDisplayName(connection.credentials)
-                    : QueryEditor.codeLensConnect,
-                command: Constants.cmdConnect,
-            }),
-        ];
-        if (connection) {
-            items.push(
+        const connection = this._connectionManager.getConnectionInfo(document.uri.toString(true));
+        if (!connection) {
+            // On no connection, show a single "Connect" CodeLens
+            return [
+                new vscode.CodeLens(new vscode.Range(0, 0, 0, 0), {
```

Comment:
While you're here, can you pull out vscode.Range(0,0,0,0) to a central spot called `TOP_OF_FILE` or something, since it's getting used a lot?